src/taskEvents/script.js

```diff
@@ -0,0 +1,284 @@
+import {
+  getTaskLogs,
+  getUserData,
+  getSelfDetails,
+  createElement,
+  getData,
+  addLoader,
+  addErrorMessage,
+  removeLoader,
+} from './utils.js';
+
+import { createEventCard } from './logCard.js';
+
+const container = document.getElementById('taskEvents-container');
+const modal = document.getElementById('modal');
+const overlay = document.querySelector('.overlay');
+
+const closeBtn = document.getElementById('closeBtn');
+closeBtn.addEventListener('click', closeModal);
+const closeBtn2 = document.getElementById('closeBtn2');
+closeBtn2.addEventListener('click', closeGenericModal);
+
+async function createProfileModal(username) {
+  try {
+    overlay.style.display = 'block';
+    const { user } = await getUserData(username);
+    // another call for roles will be made when we have userSkills collection
+    const userImg = createElement({
+      type: 'img',
+      attributes: { src: user.picture.url, alt: 'user img', class: 'userImg' },
+    });
+    const userName = createElement({
+      type: 'p',
+      attributes: { class: 'username' },
+      innerText: user.username,
+    });
+
+    document.querySelector('.top-div').prepend(userName);
+    document.querySelector('.top-div').prepend(userImg);
+
+    const skillTitle = createElement({
+      type: 'p',
+      attributes: { class: 'skillTitle' },
+      innerText: 'Skills',
+    });
+    modal.appendChild(skillTitle);
+
+    const rolesArray = [
+      'React-level1',
+      'Ember-level2',
+      'Remix-level3',
+      'NodeJs-level3',
+      'random-levl1',
+    ];
+    const roles = createRolesDiv(rolesArray);
+    modal.appendChild(roles);
+
+    const activityBtn = createUserActivityBtn(username);
+    modal.appendChild(activityBtn);
+  } catch (err) {
+    addErrorMessage(modal);
+    console.log(err);
+  }
+}
+
+function createRolesDiv(roles) {
+  const Allroles = [...roles];
```

Comment:
I am using this spread operator, so that once we are fetching the skills from the server, then if want to remove a particular skill, we can update the UI immediately and then make an API call in the background